pkg/util/commands.go

```diff
@@ -269,14 +269,17 @@ func (c *Command) run() (string, error) {
 // PathWithBinary Sets the $PATH variable. Accepts an optional slice of strings containing paths to add to $PATH
 func PathWithBinary(paths ...string) string {
 	path := os.Getenv("PATH")
-	binDir, _ := JXBinLocation()
-	answer := path + string(os.PathListSeparator) + binDir
 	mvnBinDir, _ := MavenBinaryLocation()
+	answer := ""
 	if mvnBinDir != "" {
 		answer += string(os.PathListSeparator) + mvnBinDir
 	}
 	for _, p := range paths {
-		answer += string(os.PathListSeparator) + p
+		if p != "" {
+			answer += string(os.PathListSeparator) + p
+		}
 	}
+	binDir, _ := JXBinLocation()
+	answer = binDir + string(os.PathListSeparator) + path + answer
```

Comment:
I see, so the JX_HOME/bin was not added first, but after the main PATH. That is of course wrong.